aiohttp/pytest_plugin.py

```diff
@@ -227,19 +228,22 @@ def test_client(loop):
     clients = []
 
     @asyncio.coroutine
-    def go(__param, *args, **kwargs):
-        if isinstance(__param, Application):
-            assert not args, "args should be empty"
-            client = TestClient(__param, loop=loop, **kwargs)
-        elif isinstance(__param, TestServer):
-            assert not args, "args should be empty"
-            client = TestClient(__param, loop=loop, **kwargs)
-        elif isinstance(__param, RawTestServer):
+    def go(__param, *args, server_kwargs={}, **kwargs):
+
+        if isinstance(__param, collections.Callable) and \
+                not isinstance(__param, (Application, BaseTestServer)):
+            __param = __param(loop, *args, **kwargs)
+            kwargs = {}
```

Comment:
This reproduces the exact behavior than before but a little more strict:
1. __param must be a callable, in the previous code it was the "else" case
2. kwargs is reset just after because we don't want them to be passed to the client or server